vmm/src/device_manager.rs

```diff
@@ -1818,20 +1817,16 @@ impl DeviceManager {
             ConsoleOutputMode::Pty => {
                 if let Some(pty) = serial_pty {
                     self.config.lock().unwrap().serial.file = Some(pty.path.clone());
-                    let writer = pty.main.try_clone().unwrap();
-                    let buffer = SerialBuffer::new(Box::new(writer));
                     self.serial_pty = Some(Arc::new(Mutex::new(pty)));
-                    Some(Box::new(buffer))
+                    None
                 } else {
                     let (main, mut sub, path) =
                         create_pty(true).map_err(DeviceManagerError::SerialPtyOpen)?;
                     self.set_raw_mode(&mut sub)
                         .map_err(DeviceManagerError::SetPtyRaw)?;
                     self.config.lock().unwrap().serial.file = Some(path.clone());
-                    let writer = main.try_clone().unwrap();
-                    let buffer = SerialBuffer::new(Box::new(writer));
                     self.serial_pty = Some(Arc::new(Mutex::new(PtyPair { main, sub, path })));
-                    Some(Box::new(buffer))
+                    None
```

Comment:
nit: `None` can be written only once after the if/else statement, no need to provide it for both case.